### ISSUE TITLE
wrong aws endpoint used for aws_service_discovery_private_dns_namespace

### ISSUE THREAD

**Issue by fwolfheimer** [terraform-providers/terraform-provider-aws#4457]
Hi there,

I stumbled upon a problem with aws_service_discovery_private_dns_namespace when using the eu-central-1 region. terraform is running into an endless loop trying to create this resource. Looking at the debug output, it seems that an **incorrect (non-existent) endpoint is used: servicediscovery.eu-central-1.amazonaws.com**. The AWS documentation doesn't list this endpoint, and in general there's not one such end-point for each region. For Europe only one endpoint is listed (https://docs.aws.amazon.com/general/latest/gr/rande.html): servicediscovery.eu-west-1.amazonaws.com. Indeed, if the  aws_service_discovery_private_dns_namespace is used in eu-west-1 it works as expected. 

### Terraform Version
Terraform v0.11.7
+ provider.aws v1.17.0

### Affected Resource(s)
aws_service_discovery_private_dns_namespace

### Terraform Configuration Files
```hcl
resource "aws_vpc" "example" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_service_discovery_private_dns_namespace" "example" {
  name = "hoge.example.local"
  description = "example"
  vpc = "${aws_vpc.example.id}"
}
```

### Debug Output
The following lines are repeated in the endless loop terraform enters:

2018-05-04T20:45:28.467+0200 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 2018/05/04 20:45:28 [DEBUG] [aws-sdk-go] DEBUG: Send Request servicediscovery/CreatePrivateDnsNamespace failed, will retry, error RequestError: send request failed
2018-05-04T20:45:28.467+0200 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: caused by: Post https://servicediscovery.eu-central-1.amazonaws.com/: dial tcp: lookup servicediscovery.eu-central-1.amazonaws.com on x.x.x.x:53: no such host
2018-05-04T20:45:28.467+0200 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 2018/05/04 20:45:28 [DEBUG] [aws-sdk-go] DEBUG: Retrying Request servicediscovery/CreatePrivateDnsNamespace, attempt 8
2018-05-04T20:45:28.467+0200 [DEBUG] plugin.terraform-provider-aws_v1.17.0_x4: 2018/05/04 20:45:28 [DEBUG] [aws-sdk-go] DEBUG: Request servicediscovery/CreatePrivateDnsNamespace Details:

### Expected Behavior
Resource is created

### Actual Behavior
Infinite loop

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by fwolfheimer** [comment]
Hi @bflad, thanks for your reply and thanks for pointing me to the other similar cases. Terraform being a bit more sensitive/verbose on what reply it gets back from AWS would definitely be helpful. Personally, I like to know when something unexpected happens in the backend (even if it's only a temporary network failure which goes away after a few minutes). It helps to understand an issue when it occurs. So a solution might be as simple as showing a message based on what terraform got back from AWS. Something like:
WARNING: Creation of the resource "..." failed. Error message from resource provider (AWS) was: No such host: serviceendpoint.eu-central-1.... Terraform will retry the failed operation in a few seconds.

This empowers the user of terraform to understand why an operation is failing and a he/she can decide whether this is an expected temporary failure or they can abort terraform as there's no hope for the operation to succeed. 

I think that this might improve the user experience without the need of any random timeouts and/our parsing of error messages from AWS not under control of the terraform team.

**Comment by bflad** [comment]
@fwolfheimer I would suggest taking a look at https://github.com/hashicorp/terraform/issues/17267 upstream and if it makes sense upvoting for that functionality.

**Comment by bflad** [comment]
The potential fix of lowering the retry threshold for DNS resolution errors will release with v1.18.0 of the AWS provider later today.

**Comment by bflad** [comment]
This has been released in [version 1.18.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1180-may-10-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
